### ISSUE TITLE
Option to only warn about unused bindings/namespaces when unused in all cljc branches

### ISSUE THREAD

**Issue by jwr** [clj-kondo/clj-kondo#2775]
- [X] I have read the [Clojure etiquette](https://clojure.org/community/etiquette) and will respect it when communicating on this platform.

**Is your feature request related to a problem? Please describe.**

When linting `.cljc` files with `{:cljc {:features #{:clj :cljs}}}`, clj-kondo warns about unused bindings (and namespaces) that are only used in one reader conditional branch. For example:

```
  (defn setup-handler [element callback]
    #?(:clj  (register-handler element)
       :cljs (.addEventListener element "click" callback)))
```

This warns about callback being unused, even though it is used in the `:cljs` branch. With `{:output {:langs true}}`, the warning correctly shows `[clj]`, confirming clj-kondo knows the binding is only unused in one branch.

Similarly for namespaces:

```
  (ns my-app.core
    (:require [my-app.utils :as utils]
              [my-app.platform :as platform]))

  (defn init []
    #?(:clj  (platform/start-server)
       :cljs (utils/setup-dom)))
```

Here utils is considered unused even though it's used in the `:cljs` branch, and platform is considered unused even though it's used in the `:clj` branch.

This is a common pattern in .cljc files: a function parameter or namespace is needed for one platform but has no equivalent use on the other side. The current behavior produces false positives that either need to be individually suppressed with `#_{:clj-kondo/ignore #?(:clj [:unused-binding])}` or simply ignored, which makes the linter noisy.

**Describe the solution you'd like**

A configuration option that changes the unused-binding (and unused-namespace) behavior for .cljc files to only warn when a binding is unused in all analyzed branches. Something like:

`{:cljc {:warn-unused :all-branches}}`

or scoped to individual linters:

`{:linters {:unused-binding {:cljc-mode :union}}}`

**Describe alternatives you've considered**

Adding individual ignores and ignoring entire namespaces.

**Additional context**

Why this matters to me: I have a large code base with a lot of cljc files. clj-kondo has been a wonderful tool and helped me find many cosmetic issues and several real bugs! But I cannot get rid of many warnings, which is a problem both for me and for AI tools, where spurious warnings pollute the context window. I considered marking things individually in the code, but that would take a **lot** of effort and the code would get ugly, too.

**Comment by borkdude** [comment]
It's by design that this is the default behavior since not using locals in one branch may still be something suspicious. 
But I'm open to adding an option for this.

**Comment by jwr** [comment]
I can see how this can be useful. To better explain the rationale: in my case I have lots of interactive UI components in cljc files. These have function parameters that are ignored in the :clj branch (since that branch just renders placeholders). As a result, I am flooded with clj-kondo warnings.
